src/acvp_kdf135_ssh.c

```diff
@@ -0,0 +1,389 @@
+    }
+
+    json_object_set_string(tc_rsp, "scInitIV", tmp);
+
+    tmp = calloc(1, ACVP_KDF135_SSH_MSG_MAX);
+    if (!tmp) {
+        ACVP_LOG_ERR("Unable to malloc in acvp_kdf135_ssh_output_tc");
+        return ACVP_MALLOC_FAIL;
+    }
+
+    json_object_set_string(tc_rsp, "csEKey", tmp);
+
+    tmp = calloc(1, ACVP_KDF135_SSH_MSG_MAX);
+    if (!tmp) {
+        ACVP_LOG_ERR("Unable to malloc in acvp_kdf135_ssh_output_tc");
+        return ACVP_MALLOC_FAIL;
+    }
+
+    json_object_set_string(tc_rsp, "scEKey", tmp);
+
+    tmp = calloc(1, ACVP_KDF135_SSH_MSG_MAX);
+    if (!tmp) {
+        ACVP_LOG_ERR("Unable to malloc in acvp_kdf135_ssh_output_tc");
+        return ACVP_MALLOC_FAIL;
+    }
+
+    json_object_set_string(tc_rsp, "csIKey", tmp);
+
+    tmp = calloc(1, ACVP_KDF135_SSH_MSG_MAX);
+    if (!tmp) {
+        ACVP_LOG_ERR("Unable to malloc in acvp_kdf135_ssh_output_tc");
+        return ACVP_MALLOC_FAIL;
+    }
+
+    json_object_set_string(tc_rsp, "scIKey", tmp);
+    return ACVP_SUCCESS;
+}
+
+static ACVP_RESULT acvp_kdf135_ssh_init_tc(ACVP_CTX *ctx,
+                                    ACVP_KDF135_SSH_TC *stc,
+                                    unsigned int tc_id,
+                                    ACVP_CIPHER alg_id,
+                                    unsigned int  sha_type,
+                                    unsigned int  sh_sec_len,
+                                    unsigned int  iv_len,
+                                    unsigned int  key_len,
+                                    const unsigned char *shared_sec_k,
+                                    const unsigned char *hash_h,
+                                    unsigned int hash_len,
+                                    const unsigned char *session_id,
+                                    unsigned int session_len)
+{
+    memset(stc, 0x0, sizeof(ACVP_KDF135_SSH_TC));
+
+    stc->shared_sec_k = calloc(1, ACVP_KDF135_SSH_MSG_MAX);
+    if (!stc->shared_sec_k) return ACVP_MALLOC_FAIL;
+    stc->hash_h = calloc(1, ACVP_KDF135_SSH_MSG_MAX);
+    if (!stc->hash_h) return ACVP_MALLOC_FAIL;
+    stc->session_id = calloc(1, ACVP_KDF135_SSH_MSG_MAX);
+    if (!stc->session_id) return ACVP_MALLOC_FAIL;
+
+    stc->cs_init_iv = calloc(1, ACVP_KDF135_SSH_MSG_MAX);
+    if (!stc->cs_init_iv) return ACVP_MALLOC_FAIL;
+    stc->sc_init_iv = calloc(1, ACVP_KDF135_SSH_MSG_MAX);
+    if (!stc->sc_init_iv) return ACVP_MALLOC_FAIL;
+
+    stc->cs_e_key = calloc(1, ACVP_KDF135_SSH_MSG_MAX);
+    if (!stc->cs_e_key) return ACVP_MALLOC_FAIL;
+    stc->sc_e_key = calloc(1, ACVP_KDF135_SSH_MSG_MAX);
+    if (!stc->sc_e_key) return ACVP_MALLOC_FAIL;
+
+    stc->cs_i_key = calloc(1, ACVP_KDF135_SSH_MSG_MAX);
+    if (!stc->cs_i_key) return ACVP_MALLOC_FAIL;
+    stc->sc_i_key = calloc(1, ACVP_KDF135_SSH_MSG_MAX);
+    if (!stc->sc_i_key) return ACVP_MALLOC_FAIL;
+
+    memcpy(stc->shared_sec_k, shared_sec_k, key_len);
+    memcpy(stc->hash_h, hash_h, hash_len);
+
```

Comment:
You need one for the session_id as well I assume. Also is it possible for the incoming JSON key_len property to be longer than the ACVP_KDF135_SSH_MSG_MAX. If it is we should probably add some verification of the length to be sure that we don't write passed the buffer's size.